x/stake/keeper_test.go

```diff
@@ -259,36 +275,157 @@ func TestGetValidators(t *testing.T) {
 	assert.Equal(t, candidates[3].Address, validators[1].Address, "%v", validators)
 }
 
-// TODO
 // test the mechanism which keeps track of a validator set change
 func TestGetAccUpdateValidators(t *testing.T) {
-	//TODO
+	ctx, _, keeper := createTestInput(t, nil, false, 0)
+
+	validatorsEqual := func(t *testing.T, expected []Validator, actual []Validator) {
+		require.Equal(t, len(expected), len(actual))
+		for i := 0; i < len(expected); i++ {
+			assert.Equal(t, expected[i], actual[i])
+		}
+	}
+
+	genValidators := func(candidates []Candidate) []Validator {
```

Comment:
Let's not use this at all the point of these tests was not to create the group of validators arbitrarily from the candidate list but to GET the validators from using the keeper functionality! this doesn't really provide us much value to just generate the validator set from this function